marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -216,6 +214,16 @@ private DocumentMetadataHandle buildDocumentMetadata() {
     }
 
     private String initializeJob(StructType schema) {
+        String apiPath = "/marklogic-data-hub-spark-connector/initializeJob.api";
+        String scriptPath = "/marklogic-data-hub-spark-connector/initializeJob.sjs";
+        try {
```

Comment:
I think it's worth making another private method to avoid the duplication of the "if" check and the error handling - e.g. "loadEndpointsIfNotPresent".

Reply:
Updated the method to check and load.